### DISCUSSION TITLE
Meshtastic tft-gui-work I can't compile

### DISCUSSION THREAD

**Discussion by sim1988son** [meshtastic/firmware#5289] [Troubleshooting]
**t-deck firmware-tft-gui-work** 

Meshtastic tft-gui-work t-deck in each version I get the same error. Maybe someone can help??

` *  Executing task in folder firmware-tft-gui-work: C:\Users\ameli\.platformio\penv\Scripts\platformio.exe run --environment t-deck 

Processing t-deck (board: t-deck; platform: platformio/espressif32@6.9.0; framework: arduino)
------------------------------------------------------------------------------------------------------------------------------------------------------------------------------
Verbose mode can be enabled via `-v, --verbose` option
CONFIGURATION: https://docs.platformio.org/page/boards/espressif32/t-deck.html
PLATFORM: Espressif 32 (6.9.0) > Espressif Systems LilyGO T-Deck (16 MB FLASH, 8 MB PSRAM)
HARDWARE: ESP32S3 240MHz, 320KB RAM, 16MB Flash
DEBUG: Current (esp-builtin) On-board (esp-builtin) External (cmsis-dap, esp-bridge, esp-prog, iot-bus-jtag, jlink, minimodule, olimex-arm-usb-ocd, olimex-arm-usb-ocd-h, olimex-arm-usb-tiny-h, olimex-jtag-tiny, tumpa)
PACKAGES:
 - framework-arduinoespressif32 @ 3.20017.0 (2.0.17)
 - tool-esptoolpy @ 1.40501.0 (4.5.1)
 - tool-openocd-esp32 @ 2.1100.20220706 (11.0)
 - toolchain-riscv32-esp @ 8.4.0+2021r2-patch5
 - toolchain-xtensa-esp32s3 @ 8.4.0+2021r2-patch5
LDF: Library Dependency Finder -> https://bit.ly/configure-pio-ldf
LDF Modes: Finder ~ chain, Compatibility ~ soft
[nanopb] No generation needed.
[nanopb] Installing Protocol Buffers dependencies
Requirement already satisfied: protobuf>=3.19.1 in c:\users\ameli\.platformio\penv\lib\site-packages (5.28.2)

[notice] A new release of pip is available: 24.2 -> 24.3.1
[notice] To update, run: python.exe -m pip install --upgrade pip
[nanopb] No generation needed.
Found 97 compatible libraries
Scanning dependencies...
Dependency Graph
|-- RadioLib @ 7.0.2
|-- ESP8266 and ESP32 OLED driver for SSD1306 displays @ 4.4.1+sha.e16cee1
|-- OneButton @ 2.6.1
|-- arduino-fsm @ 2.2.0+sha.7db3702
|-- TinyGPSPlus @ 1.0.3+sha.71a82db
|-- Thread @ 0.0.0+sha.1ae8778
|-- Nanopb @ 0.4.9
|-- ErriezCRC32 @ 1.0.1
|-- NonBlockingRTTTL @ 1.3.0
|-- SparkFun ATECCX08a Arduino Library @ 1.3.1+sha.5cf62b3
|-- PubSubClient @ 2.8.0
|-- NTPClient @ 3.2.1
|-- Syslog @ 2.0.0
|-- Adafruit BusIO @ 1.16.2
|-- Adafruit Unified Sensor @ 1.1.14
|-- Adafruit BMP280 Library @ 2.6.8
|-- Adafruit BMP085 Library @ 1.2.4
|-- Adafruit BME280 Library @ 2.2.4
|-- Adafruit BMP3XX Library @ 2.1.5
|-- Adafruit MCP9808 Library @ 2.0.2
|-- Adafruit INA260 Library @ 1.5.2
|-- Adafruit INA219 @ 1.2.3
|-- Adafruit MAX1704X @ 1.0.3
|-- Adafruit SHTC3 Library @ 1.0.1
|-- Adafruit LPS2X @ 2.0.6
|-- Adafruit SHT31 Library @ 2.2.2
|-- Adafruit PM25 AQI Sensor @ 1.1.1
|-- Adafruit MPU6050 @ 2.2.6
|-- Adafruit LIS3DH @ 1.3.0
|-- Adafruit AHTX0 @ 2.0.5
|-- Adafruit LSM6DS @ 4.7.3
|-- Adafruit VEML7700 Library @ 2.1.6
|-- Adafruit SHT4x Library @ 1.0.5
|-- Adafruit TSL2591 Library @ 1.4.5
|-- SparkFun Qwiic Scale NAU7802 Arduino Library @ 1.0.6
|-- SparkFun 9DoF IMU Breakout - ICM 20948 - Arduino Library @ 1.2.13
|-- ClosedCube OPT3001 @ 1.1.2
|-- EmotiBit MLX90632 @ 1.0.8
|-- DFRobot_RTU @ 1.0.3
|-- SparkFun MAX3010x Pulse and Proximity Sensor Library @ 1.1.2
|-- Adafruit MLX90614 Library @ 2.1.5
|-- bsec2 @ 1.7.2502+sha.4a08fd0
|-- BME68x Sensor library @ 1.2.40408
|-- INA3221 @ 1.0.1+sha.9964a5e
|-- SensorLib @ 0.2.0
|-- QMC5883LCompass @ 1.2.3
|-- DFRobot_LarkWeatherStation @ 1.0.0+sha.4de3a9c
|-- STK8xxx-Accelerometer @ 0.0.0+20241109013350.sha.91ceac6
|-- esp32_https_server @ 1.0.0+sha.23665b3
|-- NimBLE-Arduino @ 1.4.2
|-- libpax @ 1.1.0+sha.7bcd3fc
|-- XPowersLib @ 0.2.6
|-- ESP32 Codec2 @ 1.0.1+sha.633326c
|-- Crypto @ 0.4.0
|-- LovyanGFX @ 1.1.16
|-- bb_captouch @ 1.2.2
|-- ESP8266Audio @ 1.9.9
|-- ESP8266SAM @ 1.0.1
|-- meshtastic-device-ui @ 1.0.0
|-- lvgl @ 9.2.0+sha.0ff875423
|-- Wire @ 2.0.0
|-- SD @ 2.0.0
|-- SPI @ 2.0.0
|-- WiFi @ 2.0.0
|-- Preferences @ 2.0.0
|-- HTTPClient @ 2.0.0
|-- WiFiClientSecure @ 2.0.0
|-- WebServer @ 2.0.0
|-- ESPmDNS @ 2.0.0
|-- LittleFS @ 2.0.0
|-- EEPROM @ 2.0.0
|-- Adafruit NeoPixel @ 1.12.3
Building in release mode
fatal: not a git repository (or any of the parent directories): .git
Using meshtastic platformio-custom.py, firmware version 2.5.12 on t-deck
Linking .pio\build\t-deck\firmware.elf
.pio/libdeps/t-deck/LovyanGFX/src/lgfx/v1/LGFXBase.hpp:1360:9: warning: type 'struct LovyanGFX' violates the C++ One Definition Rule [-Wodr]
   class LovyanGFX : public
         ^
.pio/libdeps/t-deck/LovyanGFX/src/lgfx/v1/LGFXBase.hpp:1360:9: note: a type with the same name but different base type is defined in another translation unit
   class LovyanGFX : public
         ^
.pio/libdeps/t-deck/LovyanGFX/src/lgfx/v1/lgfx_filesystem_support.hpp:45:9: note: type name 'lgfx::v1::LGFX_FILESYSTEM_Support<lgfx::v1::LGFXBase>' should match type name 'lgfx::v1::LGFXBase'
   class LGFX_FILESYSTEM_Support : public Base
         ^
.pio/libdeps/t-deck/LovyanGFX/src/lgfx/v1/LGFXBase.hpp:56:9: note: the incompatible type is defined here
   class LGFXBase
         ^
.pio/libdeps/t-deck/LovyanGFX/src/lgfx/v1/LGFXBase.hpp:1375:9: warning: type 'struct LGFX_Device' violates the C++ One Definition Rule [-Wodr]
   class LGFX_Device : public LovyanGFX
         ^
.pio/libdeps/t-deck/LovyanGFX/src/lgfx/v1/LGFXBase.hpp:1375:9: note: a type with different bases is defined in another translation unit
   class LGFX_Device : public LovyanGFX
         ^
.pio/libdeps/t-deck/Crypto/AESEsp32.cpp:33:5: warning: type of 'esp_aes_setkey' does not match original declaration [-Wlto-type-mismatch]
 int esp_aes_setkey(unsigned char *ctx, const unsigned char *key,
     ^
C:/Users/ameli/.platformio/packages/framework-arduinoespressif32/tools/sdk/esp32s3/include/mbedtls/mbedtls/include/mbedtls/aes.h:184:5: note: 'esp_aes_setkey' was previously declared here
 int mbedtls_aes_setkey_dec(mbedtls_aes_context *ctx, const unsigned char *key,
     ^
C:/Users/ameli/.platformio/packages/framework-arduinoespressif32/tools/sdk/esp32s3/include/mbedtls/mbedtls/include/mbedtls/aes.h:184:5: note: code may be misoptimized unless -fno-strict-aliasing is used
.pio/libdeps/t-deck/LovyanGFX/src/lgfx/v1/LGFXBase.hpp:1378:5: warning: '__ct_base ' violates the C++ One Definition Rule  [-Wodr]
     LGFX_Device(void);
     ^
.pio/libdeps/t-deck/LovyanGFX/src/lgfx/v1/LGFXBase.cpp:3454:3: note: implicit this pointer type mismatch
   LGFX_Device::LGFX_Device(void)
   ^
.pio/libdeps/t-deck/LovyanGFX/src/lgfx/v1/LGFXBase.hpp:1375:9: note: type 'struct LGFX_Device' itself violates the C++ One Definition Rule
   class LGFX_Device : public LovyanGFX
         ^
.pio/libdeps/t-deck/LovyanGFX/src/lgfx/v1/LGFXBase.cpp:3454:3: note: '__ct_base ' was previously declared here
   LGFX_Device::LGFX_Device(void)
   ^
.pio/libdeps/t-deck/LovyanGFX/src/lgfx/v1/LGFXBase.cpp:3454:3: note: code may be misoptimized unless -fno-strict-aliasing is used
.pio/libdeps/t-deck/LovyanGFX/src/lgfx/v1/LGFXBase.hpp:1478:18: warning: 'init_impl' violates the C++ One Definition Rule  [-Wodr]
     virtual bool init_impl(bool use_reset, bool use_clear);
                  ^
.pio/libdeps/t-deck/LovyanGFX/src/lgfx/v1/LGFXBase.cpp:3477:8: note: implicit this pointer type mismatch
   bool LGFX_Device::init_impl(bool use_reset, bool use_clear)
        ^
.pio/libdeps/t-deck/LovyanGFX/src/lgfx/v1/LGFXBase.hpp:1375:9: note: type 'struct LGFX_Device' itself violates the C++ One Definition Rule
   class LGFX_Device : public LovyanGFX
         ^
.pio/libdeps/t-deck/LovyanGFX/src/lgfx/v1/LGFXBase.cpp:3477:8: note: 'init_impl' was previously declared here
   bool LGFX_Device::init_impl(bool use_reset, bool use_clear)
        ^
.pio/libdeps/t-deck/LovyanGFX/src/lgfx/v1/LGFXBase.cpp:3477:8: note: code may be misoptimized unless -fno-strict-aliasing is used
.pio/libdeps/t-deck/LovyanGFX/src/lgfx/v1/LGFXBase.hpp:1482:10: warning: 'calibrate_touch' violates the C++ One Definition Rule  [-Wodr]
     void calibrate_touch(uint16_t *parameters, uint32_t fg_rawcolor, uint32_t bg_rawcolor, uint8_t size);
          ^
.pio/libdeps/t-deck/LovyanGFX/src/lgfx/v1/LGFXBase.cpp:3528:10: note: implicit this pointer type mismatch
     void LGFX_Device::calibrate_touch(uint16_t *parameters, uint32_t fg_rawcolor, uint32_t bg_rawcolor, uint8_t size)
          ^
.pio/libdeps/t-deck/LovyanGFX/src/lgfx/v1/LGFXBase.hpp:1375:9: note: type 'struct LGFX_Device' itself violates the C++ One Definition Rule
   class LGFX_Device : public LovyanGFX
         ^
.pio/libdeps/t-deck/LovyanGFX/src/lgfx/v1/LGFXBase.cpp:3528:10: note: 'calibrate_touch' was previously declared here
     void LGFX_Device::calibrate_touch(uint16_t *parameters, uint32_t fg_rawcolor, uint32_t bg_rawcolor, uint8_t size)
          ^
.pio/libdeps/t-deck/LovyanGFX/src/lgfx/v1/LGFXBase.cpp:3528:10: note: code may be misoptimized unless -fno-strict-aliasing is used
.pio/libdeps/t-deck/LovyanGFX/src/lgfx/v1/LGFXBase.hpp:1386:10: warning: 'setPanel' violates the C++ One Definition Rule  [-Wodr]
     void setPanel(Panel_Device* panel);
          ^
.pio/libdeps/t-deck/LovyanGFX/src/lgfx/v1/LGFXBase.cpp:3469:8: note: implicit this pointer type mismatch
   void LGFX_Device::setPanel(Panel_Device* panel)
        ^
.pio/libdeps/t-deck/LovyanGFX/src/lgfx/v1/LGFXBase.hpp:1375:9: note: type 'struct LGFX_Device' itself violates the C++ One Definition Rule
   class LGFX_Device : public LovyanGFX
         ^
.pio/libdeps/t-deck/LovyanGFX/src/lgfx/v1/LGFXBase.cpp:3469:8: note: 'setPanel' was previously declared here
   void LGFX_Device::setPanel(Panel_Device* panel)
        ^
.pio/libdeps/t-deck/LovyanGFX/src/lgfx/v1/LGFXBase.cpp:3469:8: note: code may be misoptimized unless -fno-strict-aliasing is used
.pio/libdeps/t-deck/LovyanGFX/src/lgfx/v1/LGFXBase.hpp:1417:17: warning: 'setBrightness' violates the C++ One Definition Rule  [-Wodr]
     inline void setBrightness(uint8_t brightness) { _brightness = brightness; if (_panel) { _panel->setBrightness(brightness); } }
                 ^
.pio/libdeps/t-deck/LovyanGFX/src/lgfx/v1/LGFXBase.hpp:1417:17: note: implicit this pointer type mismatch
     inline void setBrightness(uint8_t brightness) { _brightness = brightness; if (_panel) { _panel->setBrightness(brightness); } }
                 ^
.pio/libdeps/t-deck/LovyanGFX/src/lgfx/v1/LGFXBase.hpp:1375:9: note: type 'struct LGFX_Device' itself violates the C++ One Definition Rule
   class LGFX_Device : public LovyanGFX
         ^
.pio/libdeps/t-deck/LovyanGFX/src/lgfx/v1/LGFXBase.hpp:1417:17: note: 'setBrightness' was previously declared here
     inline void setBrightness(uint8_t brightness) { _brightness = brightness; if (_panel) { _panel->setBrightness(brightness); } }
                 ^
.pio/libdeps/t-deck/LovyanGFX/src/lgfx/v1/LGFXBase.hpp:1417:17: note: code may be misoptimized unless -fno-strict-aliasing is used
lto-wrapper.exe: fatal error: CreateProcess: No such file or directory
compilation terminated.
c:/users/ameli/.platformio/packages/toolchain-xtensa-esp32s3/bin/../lib/gcc/xtensa-esp32s3-elf/8.4.0/../../../../xtensa-esp32s3-elf/bin/ld.exe: error: lto-wrapper failed     
collect2.exe: error: ld returned 1 exit status
*** [.pio\build\t-deck\firmware.elf] Error 1
======================================================================== [FAILED] Took 119.47 seconds ========================================================================

Environment    Status    Duration
-------------  --------  ------------
t-deck         FAILED    00:01:59.470
=================================================================== 1 failed, 0 succeeded in 00:01:59.470 =================================================================== 

 *  The terminal process "C:\Users\ameli\.platformio\penv\Scripts\platformio.exe 'run', '--environment', 't-deck'" terminated with exit code: 1. 
 *  Terminal will be reused by tasks, press any key to close it. 
`

**Comment by caveman99** [comment]
You can't compile this on windows os.

**Reply by sim1988son** [reply]
Sorry, yes on Windows os 11
